activiti-cloud-notifications-graphql-service/services/events/src/main/java/org/activiti/cloud/services/notifications/graphql/events/consumer/EngineEventsConsumerMessageHandler.java

```diff
@@ -36,33 +36,27 @@
 
     private final FluxSink<Message<List<EngineEvent>>> processorSink;
     private final Transformer transformer;
-    
+
     public EngineEventsConsumerMessageHandler(Transformer transformer,
-                                      FluxSink<Message<List<EngineEvent>>> engineEventsSink)
-    {
+                                              FluxSink<Message<List<EngineEvent>>> engineEventsSink) {
         this.processorSink = engineEventsSink;
         this.transformer = transformer;
     }
 
-    @StreamListener
-    public void receive(@Input(EngineEventsConsumerChannels.SOURCE) 
-                            Flux<Message<List<Map<String,Object>>>> input) {
-        
-        // Let's process and transform message from input stream
-        input.flatMapSequential(message -> {
-            List<Map<String, Object>> events = message.getPayload();
-            String routingKey = (String) message.getHeaders().get("routingKey");
+    @StreamListener(EngineEventsConsumerChannels.SOURCE)
+    public void receive(Message<List<Map<String, Object>>> message) {
```

Comment:
What was the reason to change the existing implementation? The existing implementation was processing events from reactive input stream using retry to handle and recover from infrastructure errors...

Reply:
The problem is that the normal implementation, used in audit and query, doesn't work if there are the reactive dependencies. The problem is a NullPointer exception:
`java.lang.NullPointerException: null
	at java.base/java.lang.Class.isAssignableFrom(Native Method)
	at org.springframework.cloud.stream.reactive.MessageChannelToFluxSenderParameterAdapter.supports(MessageChannelToFluxSenderParameterAdapter.java:40)
	at`
The exception is thrown because `type.getRawClass()` returns null when the type is `CloudRuntimeEvent[]` that is used as input of org.activiti.cloud.services.query.app.QueryConsumerChannelHandler. Now I'm looking for a solution that can preserve both dependencies without an huge refactoring.

Reply:
Replacing `CloudRuntimeEvent[]` with `List<CloudRuntimeEvent<?, ?>>` events the issue seems to disappear and after a test on my local machine, everything seems to work.